src/model_editor/InspectorGadget.cpp

```diff
@@ -789,9 +789,12 @@ void InspectorGadget::layoutComboBox(QVBoxLayout* layout, QWidget* parent, opens
   int idx = combo->findText(curVal.c_str(), Qt::MatchFixedString);
 
   if (-1 == idx) {
+
+    // field is currently set to an invalid value
     idx = 0;
-    connect(combo, static_cast<void (QComboBox::*)(const QString&)>(&QComboBox::currentTextChanged), this, &InspectorGadget::IGdefaultRemoved);
```

Comment:
The `IGdefaultRemoved` slot was only connected in this case and it was being called directly from `InspectorGadget::layoutComboBox` at line 808 ` combo->setCurrentIndex(idx)`.  This means that the slot was called before the user had a chance to do anything.